### ISSUE TITLE
Support `abstract` classes in ClassSpec

### ISSUE THREAD

**Issue by filipproch** [Arxing/dart-dartpoet#2]
There is currently no support for generating "abstract" classes as I see. Can we add it?

**Comment by filipproch** [comment]
Actually, looking at it further, all modifiers should be added (`final`, `static` etc.) for not only classes. Are you already working on it, I might want to contribute this @Arxing

**Comment by Arxing** [comment]
> There is currently no support for generating "abstract" classes as I see. Can we add it?

version 1.0.6+1 supported it

**Comment by filipproch** [comment]
@Arxing thanks for quick response, and fast release, what do you think about the support for method modifiers?

For asynchronous programming there are
 - async
- async*

Then I don't see a way to create `abstract` method (something similar to `MethodSpec` like `isAbstract`?)

**Comment by Arxing** [comment]
> Actually, looking at it further, all modifiers should be added (`final`, `static` etc.) for not only classes. Are you already working on it, I might want to contribute this @Arxing

Yes, I'm starting to upgrade dartpoet to dartpoet2, this is a big change that all specs will be related and they will construct a tree for iteration. Another change is for code generating, it will be Intuitive and  structured.

**Comment by Arxing** [comment]
> @Arxing thanks for quick response, and fast release, what do you think about the support for method modifiers?
> 
> For asynchronous programming there are
> 
> * async
> * async*
> 
> Then I don't see a way to create `abstract` method (something similar to `MethodSpec` like `isAbstract`?)

Oh! I miss it!
Now you can find `isAbstract` and `asynchronousMode` in `MethodSpec.build()` at version 1.0.6+2

**Comment by filipproch** [comment]
Thanks very much

**Comment by Arxing** [comment]
: )

**Comment by filipproch** [comment]
@Arxing do you think you could also add `const` support to `ConstructorSpec`?
